docs/development/apifirst.md

```diff
@@ -0,0 +1,118 @@
+---
+sidebar_position: 6
+---
+
+# API First REST backends with OpenAPI Specification 
+
+## Guidance (change guidance
+
+This reference describes what it means to use the API-first design approach with Node.js backends and JavaScript client applications. Reference is going to use OpenAPI standard along with supporting packages and tools we recommend to implement API first applications.
+
+> NOTE: This guide focused on the OpenAPI Specification for building RESTfull services. 
+For GraphQL please refer to the dedicated [GraphQL Guide][]
+
+### What "API-first" mean?
+
+With the API-first approach, designing the API is the first priority before writing any code. Design of the API involves thorough thinking and planning through collaboration with different teams (both client and backend side). This result in high-level documentation describing the intent of the API and ability to build client without waiting for server to be finished.
+
+This API contract acts as a central draft keeping all your team members aligned on what your API’s objectives are and how your API’s resources are exposed. The finalization of the contract allows the team to build the interface of the application.
+
+After this, the cross-functional teams rely on this interface to build the rest of the application independent of each other. In practice teams can generate backends stubs and fully functional client libraries to avoid deviation from specification set in the OpenAPI spec file.
+
+### When to use OpenAPI first and when to avoid
```

Comment:
IMHO, Contract-First is **ALWAYS** the right answer. Code first leads to re-work or mis-fitting APIs to use cases. Additionally, there are so many other advantages to Contract-First (automated test, better documentation, faster feedback, etc...) that make a code-first approach more of a "vanity" decision instead of a "technical" decision.

Reply:
Agree. Got suggestion on call to put out exactly that into details